### PR TITLE
remove conversion warnings

### PR DESCRIPTION
**Pull request by aescande**

### CODE REVIEW

File: tests/common.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ void TestMaterial::RandomTestSupportFunctionAllObjects()
 
 #ifdef DO_TEST
     for(size_t i=0; i<sObj.size(); i++)
-      supportVector[j*sObj.size() + i] = sObj[i]->support(v);
+      supportVector[j*sObj.size() + i] = sObj[static_cast<unsigned char>(i)]->support(v);
```
Comment (bchretien) [inline tests/common.cpp]: This is actually an error in sch-core, cf. [this](https://github.com/jrl-umi3218/sch-core/blob/77783063a79f459c499d4939c448663dc869d2d5/include/sch/CD/CD_Scene.h#L49).

**Comment by bchretien** [comment]
This highlights an issue in sch-core that should be solved there.

File: tests/common.cpp (hunk 1 above)
Comment (aescande) [inline tests/common.cpp]: I agree. operator[] should accept a size_t.
I'll correct this and revert part of this PR as a consequence.

**Comment by aescande** [comment]
Update the PR to remove here only the warnings that are not affect by changes in sch-core

**Pull request merged**